llama-index-core/llama_index/core/indices/document_summary/base.py

```diff
@@ -226,8 +228,10 @@ def _add_nodes_to_index(
             for node in summary_nodes:
                 node_with_embedding = node.copy()
                 node_with_embedding.embedding = id_to_embed_map[node.node_id]
+                for k in node_with_embedding.excluded_embed_metadata_keys:
+                    if node_with_embedding.metadata.get(k):
+                        node_with_embedding.metadata.pop(k)
```

Comment:
wait, why are we popping of metadata here? If they are already excluded, no need to remove?

Reply:
hmm, I was doing this so that the next step which is the vector store add, would not store the excluded kv's, but maybe that is not necessary because at first glance at the pinecone implementation it does `[node.get_content(metadata_mode=MetadataMode.EMBED)]`?

Reply:
Exactly :)

Reply:
So, I don't think the above change is needed

Reply:
done